src/approaches/interactive_learning_approach.py

```diff
@@ -21,95 +23,116 @@ class InteractiveLearningApproach(NSRTLearningApproach):
+                    self._predicates_to_learn.add(atom.predicate)
+        # Learn predicates and NSRTs.
         self._relearn_predicates_and_nsrts(dataset, online_learning_cycle=None)
-        # Track score of best atom seen so far
-        best_score = 0.0
-        # Active learning
-        for i in range(1, CFG.interactive_num_episodes + 1):
-            print(f"\nActive learning episode {i}")
-            # Sample initial state from train tasks
-            index = self._rng.choice(len(self._train_tasks))
-            state = dataset.trajectories[index].states[0]
-            # Detect and filter out static predicates
-            static_preds = utils.get_static_preds(
-                self._nsrts, self._get_current_predicates())
-            preds = self._get_current_predicates() - static_preds
-            # Find policy for exploration
-            task_list = glib_sample(state, preds, dataset)
-            assert task_list
-            task = task_list[0]
-            for task in task_list:
-                try:
-                    print("Solving for policy...")
-                    policy = self.solve(task, timeout=CFG.timeout)
-                    break
-                except (ApproachTimeout, ApproachFailure) \
-                        as e:  # pragma: no cover
-                    print(f"Approach failed to solve with error: {e}")
+        # Save dataset, to be used for online interaction.
+        self._dataset = dataset
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        # Sample a train task.
+        train_task_idx = self._rng.choice(len(self._train_tasks))
+        init = self._train_tasks[train_task_idx].init
+        # Detect and filter out static predicates.
+        static_preds = utils.get_static_preds(self._nsrts,
+                                              self._predicates_to_learn)
+        preds = self._predicates_to_learn - static_preds
+        # Find acting policy for the request.
+        task_list = glib_sample(init, preds, self._dataset)
+        assert task_list
+        task, act_policy = self._find_first_solvable(task_list)
+        assert task.init is init
+
+        def _query_policy(s: State) -> Optional[GroundAtomsHoldQuery]:
+            # Decide whether to ask about each possible atom.
+            ground_atoms = utils.all_possible_ground_atoms(
+                s, self._predicates_to_learn)
+            atoms_to_query = set()
+            for atom in ground_atoms:
+                score = score_atom(self._dataset, atom)
+                # Ask about this atom if it is the best seen so far.
+                if score > self._best_score:
+                    atoms_to_query.add(atom)
+                    self._best_score = score
+            if not atoms_to_query:
+                return None
+            return GroundAtomsHoldQuery(atoms_to_query)
+
+        def _termination_function(s: State) -> bool:
+            # Stop the episode if we reach the goal that we babbled.
```

Comment:
should we change "if" to "when"? oh nvm, because the option model could be learned/wrong in the final version of the system (just keeping this comment for clarity's sake, no action needed)